src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -246,184 +279,507 @@ private static void VerifyBytes(IntPtr p, int length)
-                // calls would randomly return previous exception objects passed to 
+                // This GetExceptionForHR call is needed to 'eat' the IErrorInfo put to TLS by
+                // Marshal.GetHRForException call above. Otherwise other Marshal.GetExceptionForHR
+                // calls would randomly return previous exception objects passed to
                 // Marshal.GetHRForException.
                 // The correct way is to use Marshal.GetHRForException at interop boundary only, but for the
                 // time being we'll keep this code as-is.
                 Marshal.GetExceptionForHR(e.HResult);
             }
         }
 
-#if netcoreapp
         [Fact]
-        public static void GenerateGuidForType()
+        public void GetHRForException_CustomException_ReturnsExpected()
+        {
+            var exception = new CustomHRException();
+            Assert.Equal(10, Marshal.GetHRForException(exception));
+        }
+
+        public class CustomHRException : Exception
+        {
+            public CustomHRException() : base()
+            {
+                HResult = 10;
+            }
+        }
+
+        [Fact]
+        public void GetHRForException_NullException_ReturnsZero()
+        {
+            Assert.Equal(0, Marshal.GetHRForException(null));
+        }
+
+        public static IEnumerable<object[]> GenerateGuidForType_Valid_TestData()
+        {
+            yield return new object[] { typeof(int) };
+            yield return new object[] { typeof(int).MakePointerType() };
+            //yield return new object[] { typeof(int).MakeByRefType() };
+            yield return new object[] { typeof(string) };
+            yield return new object[] { typeof(string[]) };
+
+            yield return new object[] { typeof(NonGenericClass) };
+            yield return new object[] { typeof(GenericClass<string>) };
+            yield return new object[] { typeof(AbstractClass) };
+
+            yield return new object[] { typeof(NonGenericStruct) };
+            yield return new object[] { typeof(GenericStruct<string>) };
+
+            yield return new object[] { typeof(NonGenericInterface) };
+            yield return new object[] { typeof(GenericInterface<string>) };
+
+            yield return new object[] { typeof(GenericClass<>) };
+            //yield return new object[] { typeof(GenericClass<>).GetTypeInfo().GenericTypeParameters[0] };
+
+            yield return new object[] { typeof(ClassWithGuidAttribute) };
+        }
+
+        [Theory]
+        [MemberData(nameof(GenerateGuidForType_Valid_TestData))]
+        public void GenerateGuidForType_ValidType_ReturnsExpected(Type type)
+        {
+            if (type.HasElementType)
+            {
+                if (PlatformDetection.IsNetCore)
+                {
+                    Assert.Equal(Guid.Empty, type.GUID);
+                    Assert.Equal(type.GUID, Marshal.GenerateGuidForType(type));
```

Comment:
Could you create an issue for this behavior difference between NetCore and other platform?

Reply:
Opened https://github.com/dotnet/corefx/issues/30940